From c52e8734d043ed43d21370b2c29104fdc8cc8cb7 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 16 Apr 2025 19:51:01 -0700
Subject: [PATCH] Propagate -ffile-prefix-map from CFLAGS to ASFLAGS.

-ffile-prefix-map also implies -fdebug-prefix-map
and some systems may use --ffile-prefix-map for building
reprodubile systems

Upstream-Status: Submitted [https://patchwork.sourceware.org/project/glibc/patch/20250429064632.2048954-1-raj.khem@gmail.com/]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makeconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makeconfig b/Makeconfig
index 603ed955c4..2f0e5d25f8 100644
--- a/Makeconfig
+++ b/Makeconfig
@@ -1194,7 +1194,7 @@ endif
 
 # The assembler can generate debug information too.
 ifndef ASFLAGS
-ASFLAGS := $(filter -g% -fdebug-prefix-map=% -m%,$(CFLAGS))
+ASFLAGS := $(filter -g% -fdebug-prefix-map=% -ffile-prefix-map=% -m%,$(CFLAGS))
 endif
 override ASFLAGS += -Werror=undef $(ASFLAGS-config) $(asflags-cpu) $(as-sframe)
 
